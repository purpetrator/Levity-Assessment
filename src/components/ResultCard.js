import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";
import { monthDayYear } from "../utils";

export function ResultCard(props) {
  return (
    <div className="card">
      <div className="card-body">
        <div className="card-title">
          <img className="owner-avatar" src={props.avatar}  alt={"avatar"}/>
          <a className="repo-owner" href={props.repoURL} target="_blank">
            {props.name}
          </a>
        </div>
        <div className="star-count">
          <FontAwesomeIcon className="star-icon" icon={faStar} />
          {props.count}
        </div>
        <p className="card-text">{props.description}</p>
        <div className="created-at">
          Created on {monthDayYear(props.createdAt)}
        </div>
      </div>
    </div>
  );
}

export default ResultCard;
