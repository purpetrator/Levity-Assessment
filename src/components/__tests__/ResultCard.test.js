import React from "react";
import TestRenderer from "react-test-renderer";
import { ResultCard } from "../ResultCard";

it("successful renders the component", () => {
    const component = TestRenderer.create(
            <ResultCard
                name={"Ana Banana"}
                count={50000234}
                description={"A really fun app!"}
                avatar={"https://fake.com/photoUrl"}
                repoURL={"https://fake.com"}
                createdAt={"2017-09-27T15:31:23Z"}
            />
    );

    expect(component).toMatchSnapshot();
});
