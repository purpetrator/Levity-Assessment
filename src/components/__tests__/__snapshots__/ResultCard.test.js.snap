// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`successful renders the component 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title"
    >
      <img
        alt="avatar"
        className="owner-avatar"
        src="https://fake.com/photoUrl"
      />
      <a
        className="repo-owner"
        href="https://fake.com"
        target="_blank"
      >
        Ana Banana
      </a>
    </div>
    <div
      className="star-count"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-star star-icon"
        data-icon="star"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      50000234
    </div>
    <p
      className="card-text"
    >
      A really fun app!
    </p>
    <div
      className="created-at"
    >
      Created on 
      9/27/2017
    </div>
  </div>
</div>
`;
